code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -140,7 +140,8 @@
 			return
 
 	if(target && loc == target.loc)
-		clean(target)
+		if(!check_bot(target))	//Target is not defined at the parent
```

Comment:
I'm only checking for floorbots and cleanbots because they are the only ones who really suffer from stacking.